pkg/storage/chunk/client/azure/blob_storage_client_test.go

```diff
@@ -24,6 +28,58 @@ func (fn RoundTripperFunc) RoundTrip(req *http.Request) (*http.Response, error)
 	return fn(req)
 }
 
+type FederatedTokenTestSuite struct {
+	suite.Suite
+	config                      *BlobStorage
+	mockOAuthConfig             *adal.OAuthConfig
+	mockedServicePrincipalToken *adal.ServicePrincipalToken
+}
+
+func (suite *FederatedTokenTestSuite) SetupTest() {
```

Comment:
Interesting. Are you using the suite to get a test setup code?

Reply:
I use the suite to set up an isolated environment with mocks for testing the Azure federated token functionality. There might be other ways to do it, but I come from the Ruby/Rspec world, and using a suite was the most familiar approach for me.